x-pack/filebeat/input/awss3/sqs_s3_event.go

```diff
@@ -152,15 +166,13 @@ func (p *sqsS3EventProcessor) ProcessSQS(ctx context.Context, msg *types.Message
 		return nil
 	}
 
-	if p.maxReceiveCount > 0 && !errors.Is(processingErr, &nonRetryableError{}) {
+	if p.maxReceiveCount > 0 && !errors.Is(processingErr, &nonRetryableError{}) && hasReceiveCountAttribute {
 		// Prevent poison pill messages from consuming all workers. Check how
 		// many times this message has been received before making a disposition.
-		if v, found := msg.Attributes[sqsApproximateReceiveCountAttribute]; found {
-			if receiveCount, err := strconv.Atoi(v); err == nil && receiveCount >= p.maxReceiveCount {
-				processingErr = nonRetryableErrorWrap(fmt.Errorf(
-					"sqs ApproximateReceiveCount <%v> exceeds threshold %v: %w",
-					receiveCount, p.maxReceiveCount, processingErr))
-			}
+		if receiveCount, err := strconv.Atoi(rawRecieveCount); err == nil && receiveCount >= p.maxReceiveCount {
```

Comment:
I think this could be refactored to avoid multiple calls to atoi on `rawRecieveCount`.